File: benchmarks/arithmetic.go
package main

import (
	"fmt"
	"time"
)

const runs = 3

func benchmarkArithmetic(iterations int64) int64 {
	var sum int64 = 0
	for i := int64(0); i < iterations; i++ {
		sum = ((sum + i) * i + 3) % 7
	}
	return sum
}

func fib(n int64) int64 {
	if n < 2 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	fmt.Println("=== Go Benchmarks ===")

	// Benchmark 1: Arithmetic loop
	iterations := int64(10000000)
	benchmarkArithmetic(iterations) // warmup
	var best int64 = 1 << 62
	var result int64
	for run := 0; run < runs; run++ {
		start := time.Now()
		r := benchmarkArithmetic(iterations)
		elapsed := time.Since(start).Nanoseconds()
		if elapsed < best {
			best = elapsed
			result = r
		}
	}
	fmt.Printf("Arithmetic loop (%d iterations):\n", iterations)
	fmt.Printf("  Time: %d ms\n", best/1000000)
	fmt.Printf("  Result: %d\n", result)

	// Benchmark 2: Recursive fibonacci
	n := int64(35)
	fib(n) // warmup
	best = 1 << 62
	for run := 0; run < runs; run++ {
		start := time.Now()
		r := fib(n)
		elapsed := time.Since(start).Nanoseconds()
		if elapsed < best {
			best = elapsed
			result = r
		}
	}
	fmt.Printf("Fibonacci (n=%d):\n", n)
	fmt.Printf("  Time: %d ms\n", best/1000000)
	fmt.Printf("  Result: %d\n", result)
}

File: benchmarks/arithmetic.rs
use std::hint::black_box;
use std::time::Instant;

const RUNS: usize = 3;

fn benchmark_arithmetic(iterations: i64) -> i64 {
    let mut sum: i64 = 0;
    for i in 0..iterations {
        sum = ((sum + i) * i + 3) % 7;
    }
    sum
}

fn fib(n: i64) -> i64 {
    if n < 2 {
        n
    } else {
        fib(n - 1) + fib(n - 2)
    }
}

fn main() {
    println!("=== Rust Benchmarks ===");

    // Benchmark 1: Arithmetic loop
    let iterations = 10_000_000i64;
    let _ = benchmark_arithmetic(black_box(iterations)); // warmup
    let mut best = u128::MAX;
    let mut result: i64 = 0;
    for _ in 0..RUNS {
        let start = Instant::now();
        let r = benchmark_arithmetic(black_box(iterations));
        let elapsed = start.elapsed().as_nanos();
        if elapsed < best { best = elapsed; result = r; }
    }
    println!("Arithmetic loop ({} iterations):", iterations);
    println!("  Time: {} ms", best / 1_000_000);
    println!("  Result: {}", result);

    // Benchmark 2: Recursive fibonacci
    let n = 35i64;
    let _ = fib(black_box(n)); // warmup
    best = u128::MAX;
    for _ in 0..RUNS {
        let start = Instant::now();
        let r = fib(black_box(n));
        let elapsed = start.elapsed().as_nanos();
        if elapsed < best { best = elapsed; result = r; }
    }
    println!("Fibonacci (n={}):", n);
    println!("  Time: {} ms", best / 1_000_000);
    println!("  Result: {}", result);
}
